### DISCUSSION TITLE
Wondering about the lack of pattern, batches and reps arguments for tar_stan_mcmc

### DISCUSSION THREAD

**Discussion by lucumas** [wlandau/stantargets#48] [Help] [answered]
I've been building a pipeline with plain `targets`, using the `pattern` argument to map over datasets generated in a previous target. I'm now trying to simplify the code using `stantargets`, but I've noticed none of the functions seem to take a `pattern` argument.

I've already figured out a way to adapt the pipeline using `tar_map` and the `batches` and `reps` arguments, but I've noticed `tar_stan_mcmc` doesn't accept those either.

Mostly I'd just like to understand these choices as they seem a bit haphazard though I'm sure there's some underlying reason for them.

**Comment by wlandau** [comment, marked as answer]
`tar_stan_mcmc()` deliberately blocks dynamic branching. The intent is to analyze one dataset with a small number of models and create targets for each of the output types for each model: the cmdstan fit object, summaries, draws, and HMC diagnostics.

`tar_stan_mcmc_rep_sumamry()` blocks `pattern` so there is less room for user error with dynamic branching. The only dynamic branching is tightly controlled through batches and reps.

Target factories aim to simplify the user experience for well-understood domain-specific workflows such as Bayesian data analysis. Part of this is abstracting away dynamic branching so users do not need to think about it. https://ropensci.org/blog/2021/02/03/targets/.

**Comment by dmi3kno** [comment]
So the suggestion is to go back to `tar_target()` and build your own targets for file, fit, summaries etc (instead of using `stantargets`? I am interested in running one model per fraction of the data in independent branches.

The reason I can not do replications is because each of the replications consists of 5 bayesian models executed sequentially, so my whole pipeline is N branches with 5 models executed sequentially in each branch. Each model is, of course, 4 chains, 10000 iterations ;(

Found this [repo](https://github.com/wlandau/targets-stan). Thanks for good examples!
